### ISSUE TITLE
[Feature] hardware keylogger/screenshot resistant wallet restore entry method for cli

### ISSUE THREAD

**Issue by nasaWelder** [monero-project/monero#2869]
If one of [keyboard|monitor] is physically compromised for an airgapped/liveUSB/Tails cold wallet (or for warm wallet: a non-targeted keylogger only virus), current wallet restore methods expose seed to attacker.

I realize this is very hypothetical at this point, but if a South African does ever have to flee, restoring wallet at refugee camp could have compromised keyboard...

demo:
https://github.com/nasaWelder/sandbox/tree/master/monero

I only know python, so my prototype is just a demonstration at this point. It only relies on standard library for 2.7+,3.0+ so I hope that encourages testing for yourself. It also wipes screen at exit so terminal does not show the seed if someone scrolls up. If no one picks this up It could be a good opportunity for me to learn C++.

Missing from demo:
1) at exit, public address should be displayed for user to check that input was error free, with confirmation dialogue.

If this is too trivial, I apologize. I'm new to contributing/open-source in general.

**Comment by moneromooo-monero** [comment]
Sounds like something that'd be even more useful for the GUI.

**Comment by nasaWelder** [comment]
I bet you could use cli version as back end for gui. I make guis in python IRL. Python is so good at error Handling, I wish the gui was in that, would be so easy to contribute.

**Comment by dEBRUYNE-1** [comment]
@nasaWelder It's not that difficult as you think probably. This might be of help:

https://blog.cedsys.nl/misc-monero-gui-development-intro.html

**Comment by dEBRUYNE-1** [comment]
+proposal

**Comment by fresheneesz** [comment]
I don't think this would materially improve the security of your monero. If someone can keylog, they can also screen capture and find out what you saw on the screen when you were typing or mouse clicking. Monero just needs hardware wallets, that'll solve the problem.

**Comment by moneromooo-monero** [comment]
+hacktoberfest

**Comment by selsta** [comment]
Closing, this doesn't appear like something we will implement.
